junit-jupiter-api/src/main/java/org/junit/jupiter/api/MethodOrderer.java

```diff
@@ -172,6 +175,76 @@ private static int getOrder(MethodDescriptor descriptor) {
 		}
 	}
 
+	/**
+	 * {@code MethodOrderer} that sorts methods based on the {@link Depend @Depend}
+	 * annotation.
+	 */
+	class DependAnnotation implements MethodOrderer {
+		private static final Logger logger = LoggerFactory.getLogger(Random.class);
+
+		private static Map<String, Integer> dependencySize;
+		private static Map<String, String[]> digraph;
```

Comment:
A `MethodOrderer` must not have `static` state. Please redesign this.

Reply:
I have remove all these class variable except for Logger and provide them as parameters to depthFirstSearch.